### PR TITLE
Use pyproject.toml only (move away from setup.cfg)

### LINKED ISSUE #1505: Transition Woodwork to use pyproject.toml only (move away from setup.cfg)

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #1505
- One thing we might want to consider before merging this, I got the following warning when building the wheel & tar.gz files:
```python
/tmp/build-env-o6cmnata/lib/python3.8/site-packages/setuptools/config/pyprojecttoml.py:108: 
_BetaConfiguration: 
Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
  warnings.warn(msg, _BetaConfiguration)
```
- https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html?highlight=declaring%20configurations%20not%20standardized%20by#setuptools-specific-configuration

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Future Release
     * Enhancements
     * Fixes
     * Changes
+        * Transition from setup.cfg to pyproject.toml (:pr:`778`)
```
Comment (gsheni) [inline docs/source/release_notes.rst:11]:
```suggestion
        * Transition from setup.cfg to pyproject.toml (:pr:`1506`)
```

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[flake8]
```
Comment (gsheni) [inline .flake8:1]: > The discussion of the original proposal ([#234](https://github.com/PyCQA/flake8/issues/234)) was closed as "too heated", subsequent feature and pull requests were marked as "spam" ([#1332](https://github.com/PyCQA/flake8/pull/1332), [#1421](https://github.com/PyCQA/flake8/issues/1421), [#1431](https://github.com/PyCQA/flake8/issues/1431), [#1447](https://github.com/PyCQA/flake8/issues/1447), [#1501](https://github.com/PyCQA/flake8/issues/1501)).

Comment (gsheni) [reply .flake8:1]: flake8 refuses to support pyproject.toml (see below) and therefore we are using this config file.

> The discussion of the original proposal ([#234](https://github.com/PyCQA/flake8/issues/234)) was closed as "too heated", subsequent feature and pull requests were marked as "spam" ([#1332](https://github.com/PyCQA/flake8/pull/1332), [#1421](https://github.com/PyCQA/flake8/issues/1421), [#1431](https://github.com/PyCQA/flake8/issues/1431), [#1447](https://github.com/PyCQA/flake8/issues/1447), [#1501](https://github.com/PyCQA/flake8/issues/1501)).

File: pyproject.toml [hunk 3]
```diff
@@ -1,10 +1,133 @@
...
+dynamic = ["version"]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development",
+    "Topic :: Scientific/Engineering",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.7",
```
Comment (gsheni) [inline pyproject.toml:21]:
```suggestion
```

**Review by cp2boston — COMMENTED** [review]
This is great!  I ran the pre-commits and tests on a clean repo as a simple check

**Review by bchen1116 — APPROVED** [review]
LGTM! Thanks for doing this. I'm not too worried about the warning of the feature being in beta, so that shouldn't be a blocker for merging.

**Pull request merged**